EventsExpress/ClientApp/src/components/event/bookmarks/bookmark-button.js

```diff
@@ -0,0 +1,40 @@
+import Tooltip from '@material-ui/core/Tooltip';
+import IconButton from '@material-ui/core/IconButton';
+import React from 'react';
+import { connect } from 'react-redux';
+import { deleteBookmark, saveBookmark } from '../../../actions/bookmarks/event-bookmarks-actions';
+
+const BookmarkButton = ({ event, currentUser, ...props }) => {
+    const bookmarked = currentUser.bookmarkedEvents.includes(event.id);
+
+    const toggleBookmark = () => {
+        if (bookmarked) {
+            props.deleteBookmark(event.id);
+        } else {
+            props.saveBookmark(event.id);
+        }
+    };
+
+    return (
+        <>
+            {currentUser.id && (
+                <Tooltip title="Bookmark">
+                    <IconButton key={+bookmarked} onClick={toggleBookmark}>
```

Comment:
Omg, that is a hack :)

Reply:
I had to add this because of Icon issue, without the key it won't re-render